aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -726,7 +726,11 @@ Tensor& add_out_dense_sparse_cpu(Tensor& r, const Tensor& dense, const SparseTen
 
 Tensor mul_sparse(const Tensor& self, const Tensor& other) {
   auto commonDtype = at::result_type(self, other);
-  Tensor result = at::empty({0}, self.options().dtype(commonDtype));
+  // This code path is also hit when doing `0dim-Tensor * other`, which can
+  // be mul(dense, sparse). Make sure we use the sparse exemplar for result.
```

Comment:
Thanks for the update! Sorry to be so nit-picky. This is saying *when* this is being called, but I'm wondering *why* this needs to happen in the first place. If it's being called like this, what is it that necessitates this divergent behavior for sparse only.

Reply:
There is no (dense, sparse) nor (sparse, dense) multiplication path. The code leading to here lets non-tensor objects and 0dim tensors through and blocks general (dense, sparse) multiplication. I am not sure what text would be satisfactory, could you suggest something?

Reply:
@mattip - what you just wrote is a good explanation. So I'd write something like
```
mul_sparse currently doesn't support (dense, sparse) or (sparse, dense) multiplication but we can still enable it for 0dim tensors by [...]
```
Non-tensor object doesn't seem to apply since both arguments are Tensors. What do you mean by non-tensor object?

Reply:
`2 * a` (where `a` is a sparse Tensor) will end up here too, because `2` will be turned into a Tensor

Reply:
updated the comment.

Reply:
Yes, but that cast happens at the level of the bindings / factory functions and will turn into a 0-dim Tensor as well. It's also in contrast to the explicit Scalar overloads we have for some functions.